ecosystem/sep-0043.md

```diff
@@ -0,0 +1,146 @@
+Track: Standard
+Status: Draft
+Created: 2024-04-11
+Version: 1.2.0
+Discussion: https://github.com/stellar/stellar-protocol/discussions/1467
+```
+
+## Abstract
+
+This standard aims to enhance developer experiences by standardizing an interface for web wallet's to use when
+interacting with other codebases.
+
+## Motivation
+
+There are numerous Stellar wallets in the ecosystem with this number expected to grow as the ecosystem does. Wallets
+expose an API to allow dapps to call methods that preform such common tasks as transmitting a wallet's address. An issue
+is that wallets all use slightly different API's to preform similar functions, with different method names and call
+signature as well as different return values. Developers have a motivation to integrate with as many as possible, but
+currently, they must create abstractions to handle every wallet they wish to connect to. This proposal presents a
+standard API interface for all wallets to implement. If all wallets have the same interface, it removes the needs for
+developers to create abstractions for every individual wallet they want to integrate with. Developers can safely assume
+a wallet uses this agreed upon interface, creating simpler codepaths that easily connect to current wallets while also
+simplifying the onboarding of new wallets.
+
+## Specification
+
+The standard web wallet interface provides a structure for common methods that wallets are expected to offer. In
+addition, the interface defines common error message shapes to ensure wallets adhere to informative and usable error
+messaging. Messages are designed to be surfaced directly to the user, so they should provide a readable description of
+what went wrong.
+
+### Wallet Interface Format
+
+Wallets will employ an API using the following format:
+
+```typescript
+
+interface Error {
+  message: string,    // general description message returned to the client app
+  code: number,       // unique error code
+  ext?: Array<string>  // optional extended details
+}
+
+{
+  getAddress: () => Promise<{address: string;} | Error>,
+
+  signTransaction: (xdr: string, opts?: { networkPassphrase?: string, address?: string; submit?: boolean; submitUrl?: string; }) => Promise<{signedTxXdr: string; signerAddress: string;} | Error>,
+
+  signAuthEntry: (authEntry: string, opts?: { networkPassphrase?: string, address?: string }) => Promise<{signedAuthEntry: string; signerAddress: string;} | Error>,
+
+  signMessage: (message: string, opts?: { networkPassphrase?: string, address?: string }) => Promise<{signedMessage: string; signerAddress: string;} | Error>,
+
+  getNetwork: () => Promise<{network: string, networkPassphrase: string}>
+}
+```
+
+In addition, this proposal has a standard for error interfaces. Below are examples of different kinds of errors a wallet
+may surface:
```

Comment:
We should heavily emphasize which parts of the errors are standard (i.e. the interface or categories) vs. examples that wallets should fill in themselves (i.e. the messages/codes themselves, mentioning localization etc. might be worthwhile), because it's not super clear to me here which ones are "guaranteed" for an implementor.